bcda/cclf/testutils/cclfUtils.go

```diff
@@ -67,15 +66,20 @@ func ImportCCLFPackage(acoSize, environment string) (err error) {
 		//timeStr := time.Now().Add(time.Minute * time.Duration(i-1)).Format("1504059")
 		archiveName = fmt.Sprintf("T.BCD.A%s.ZCY%s.T%s", acoIDNum, dateStr, "0000000")
 		filename := fmt.Sprintf("T.BCD.A%s.%sY%s.T%s", acoIDNum, file.Name(), dateStr, "0000000")
-		sourceFile_fileName := fmt.Sprintf("%s/%s__%s", sourcedir, file.Name(),filename)
-		fileList = append(fileList, sourceFile_fileName)
+		sourceFilename := fmt.Sprintf("%s/%s__%s", sourcedir, file.Name(), filename)
+		fileList = append(fileList, sourceFilename)
 	}
 
 	newZipFile, err := os.Create(fmt.Sprintf("%s/%s", DestDir, archiveName))
 	if err != nil {
 		return err
 	}
-	defer newZipFile.Close() // #nosec G307
+	defer func() {
+		if ferr := newZipFile.Close(); ferr != nil {
+			err = ferr
```

Comment:
Should we at least log the error?

Reply:
the error isn't being ignored, but it is overwriting any error that is being returned by the outer function.

I'm going to rewrite this to use the helper method.

Reply:
This is testing my golang skills. The logic here is checking if the error exists and if the returned error is nil, and then setting the named return value to the error. This is to prevent overwriting any other error being returned by the method.